management/branding.go

```diff
@@ -16,10 +20,102 @@ type Branding struct {
 type BrandingColors struct {
 	// Accent color
 	Primary *string `json:"primary,omitempty"`
+
 	// Page background color
+	// Only one of PageBackground and PageBackgroundGradient should be set.
+	// If both fields are set, PageBackground takes priority.
+	PageBackground *string `json:"-"`
+
+	// Page background gradient
+	// Only one of PageBackground and PageBackgroundGradient should be set.
+	// If both fields are set, PageBackground takes priority.
+	PageBackgroundGradient *BrandingPageBackgroundGradient `json:"-"`
+}
+
+type BrandingPageBackgroundGradient struct {
+	Type        *string `json:"type,omitempty"`
+	Start       *string `json:"start,omitempty"`
+	End         *string `json:"end,omitempty"`
+	AngleDegree *int    `json:"angle_deg,omitempty"`
+}
+
+type jsonPageBackgroundSolid struct {
 	PageBackground *string `json:"page_background,omitempty"`
 }
 
+type jsonPageBackgroundGradient struct {
+	PageBackgroundGradient *BrandingPageBackgroundGradient `json:"page_background,omitempty"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+//
+// It is required to handle the json field page_background, which can either
+// be a hex color string, or an object describing a gradient.
+func (bc *BrandingColors) MarshalJSON() ([]byte, error) {
+	var data interface{}
```

Comment:
Using `interface{}` and inline struct definitions feels non-idiomatic, do you have any suggestions here @alexkappa ?

Reply:
This was resolved by using an embedded type for all normal fields of `BrandingColor`, and an `interface{}` for variable `page_background` field.

See https://github.com/go-auth0/auth0/pull/99#discussion_r403211848